bundles/wps/wps-service-scripts/src/test/java/org/orbisgis/wpsservicescripts/OrbisGISWpsScriptPluginTest.java

```diff
@@ -0,0 +1,235 @@
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Class to test the well working of the plugin life cycle :
+ *
+ * On activating : add all the script of the resource folder to the WPS server.
+ *
+ * On deactivating : remove all the script add on activating.
+ *
+ * @author Sylvain PALOMINOS
+ */
+public class OrbisGISWpsScriptPluginTest {
+
+    /**
+     * Test the life cycle of the plugin.
+     */
+    @Test
+    public void testPluginLifeCycle(){
+        //Gets the list of the script files add by the plugin
+        CustomWpsService localWpsServer = new CustomWpsService();
+        CustomCoreWorkspace customCoreWorkspace = new CustomCoreWorkspace();
+        Assert.assertNotNull("Unable to create the temporary folder for the custom CoreWorkspace.",
+                customCoreWorkspace.getApplicationFolder());
+        OrbisGISWpsScriptPlugin plugin = new OrbisGISWpsScriptPlugin();
+        plugin.setLocalWpsService(localWpsServer);
+        plugin.setCoreWorkspace(new CustomCoreWorkspace());
+        plugin.activate();
+        List<File> addScriptList = localWpsServer.getAddScriptList();
+        //Gets the list of the script files contained in the resource folder of the plugin
+        File folder = new File(this.getClass().getResource("scripts").getFile());
+        List<File> resourceGroovyScriptList = getAllGroovyScripts(folder);
+        //Test if the resource files are add by the plugin
+        for(File resourceScript : resourceGroovyScriptList){
+            boolean isResourceScriptAdd = false;
+            for(File addScript : addScriptList){
+                if(resourceScript.getName().equals(addScript.getName())){
+                    isResourceScriptAdd = true;
+                }
+            }
+            Assert.assertTrue("The resource file '"+resourceScript.getName()+"' should be add by the plugin.", isResourceScriptAdd);
+        }
+        //Deactivate the plugin
+        plugin.deactivate();
+        //Test if all the script are removed
+        Assert.assertTrue("All the scripts should have been removed from the server.",
+                localWpsServer.getAddScriptList().isEmpty());
+        plugin.unsetCoreWorkspace(null);
+        plugin.unsetInternalWpsClient(null);
+    }
+
+    /**
+     * Returns the list of the groovy script file of a given directory.
+     * @param folder Directory to explore.
+     * @return The list of the groovy script files.
+     */
+    private List<File> getAllGroovyScripts(File folder){
+        List<File> scriptList = new ArrayList<>();
+        for(File f : folder.listFiles()) {
+            if (f.isDirectory()) {
+                scriptList.addAll(getAllGroovyScripts(f));
+            }
+            else {
+                scriptList.add(f);
+            }
+        }
+        return scriptList;
+    }
+
+    /**
+     * LocalWpsServer implementation used to check if all the scripts add by the plugin are in the resources and if all
```

Comment:
Yes it uses the script in `src/main/resources/org/orbisgis/wpsservicescripts/scripts`.